sklearn/datasets/tests/test_kddcup99.py

```diff
@@ -44,3 +44,35 @@ def test_shuffle(fetch_kddcup99_fxt):
     dataset = fetch_kddcup99_fxt(random_state=0, subset='SA', shuffle=True,
                                  percent10=True)
     assert(any(dataset.target[-100:] == b'normal.'))
+
+
+def test_asframe(fetch_kddcup99_fxt):
+    pd = pytest.importorskip('pandas')
+
+    data = fetch_kddcup99_fxt(as_frame=True)
+    assert hasattr(data, 'frame') is True
+    assert isinstance(data.frame, pd.DataFrame)
```

Comment:
Regarding the type and dtype, you can import the function `check_as_frame` defined in `test_common.py` which should be able to check those. Once that we have several fetcher compatible, we could make a common test directly.

Reply:
DONE.